backend/replicache-transaction.ts

```diff
@@ -54,15 +63,38 @@ export class ReplicacheTransaction implements WriteTransaction {
     return val !== undefined;
   }
 
-  // TODO!
   async isEmpty(): Promise<boolean> {
-    throw new Error("not implemented");
-  }
-  scan(): ScanResult<string> {
-    throw new Error("not implemented");
+    for await (const _ of this.scan()) {
+      return false;
+    }
+    return true;
   }
-  scanAll(): Promise<[string, JSONValue][]> {
-    throw new Error("not implemented");
+
+  scan(options: ScanOptions = {} as ScanNoIndexOptions) {
+    if (isScanIndexOptions(options)) {
+      throw new Error("not implemented");
+    }
+
+    const { _executor: executor, _spaceID: spaceID, _cache: cache } = this;
+
+    return makeScanResult(options, async function* (fromKey) {
+      const source = getEntries(executor, spaceID, fromKey);
+
+      // TODO: It would be more optimal to keep the _cache in a sorted map in
+      // the first place.
+      const changes = [...cache]
+        .filter(([, { dirty }]) => dirty)
+        .filter(([k]) => k.localeCompare(fromKey) >= 0)
+        .map(([k, { value }]) => [k, value] as const)
+        .sort(([k1], [k2]) => k1.localeCompare(k2));
```

Comment:
You are probably better of doing a for of loop here so you do not have to allocate 3 extra arrays